sys/psa_crypto/psa_hmac.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2024 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto
+ * @defgroup    sys_psa_crypto_hmac  PSA Generic HMAC
+ * @{
+ *
+ * @file        psa_hmac.c
+ * @brief       Generic PSA Crypto HMAC implementation.
+ *
+ * @author      Armin Wolf <armin.wolf@mailbox.tu-dresden.de>
+ *
+ * @}
+ */
+
+#include "kernel_defines.h"
+#include "psa/crypto.h"
+#include "psa/crypto_contexts.h"
+#include "psa_hmac.h"
+#include "string_utils.h"
+
+#if IS_USED(MODULE_PSA_RIOT_MAC_HMAC_GENERIC)
+static psa_status_t psa_hmac_setup_key(psa_mac_operation_t *operation,
+                                       const uint8_t *key_buffer,
+                                       size_t key_buffer_size)
+{
+    psa_algorithm_t hash_alg = PSA_ALG_GET_HASH(operation->alg);
+    size_t block_length = PSA_HASH_BLOCK_LENGTH(hash_alg);
+    size_t dummy;
+
+    if (key_buffer_size > block_length) {
+        return psa_hash_compute(hash_alg, key_buffer, key_buffer_size, operation->block,
+                                block_length, &dummy);
+    }
+
+    memcpy(operation->block, key_buffer, key_buffer_size);
+
+    return PSA_SUCCESS;
+}
+
+static psa_status_t psa_mac_finish(psa_mac_operation_t *operation,
+                                   uint8_t *buffer,
+                                   size_t buffer_size)
+{
+    psa_algorithm_t hash_alg = PSA_ALG_GET_HASH(operation->alg);
+    size_t block_length = PSA_HASH_BLOCK_LENGTH(hash_alg);
+    psa_status_t status;
+    size_t hash_length;
+
+    status = psa_hash_finish(&operation->hash, buffer, buffer_size, &hash_length);
+    if (status != PSA_SUCCESS) {
+        return status;
+    }
+
+    for (size_t i = 0; i < block_length; i++) {
+        operation->block[i] = operation->block[i] ^ 0x5c;
+    }
+
+    status = psa_hash_setup(&operation->hash, hash_alg);
+    if (status != PSA_SUCCESS) {
+        return status;
+    }
+
+    status = psa_hash_update(&operation->hash, operation->block, block_length);
+    if (status != PSA_SUCCESS) {
+        return status;
+    }
+
+    status = psa_hash_update(&operation->hash, buffer, hash_length);
+    if (status != PSA_SUCCESS) {
+        return status;
+    }
+
+    // Clear key data
```

Comment:
```suggestion
    /* Clear key data */
```

please use C-style comments as per the [coding convention](https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md#comments)

Reply:
Will do.